### PR TITLE
docs: add information on how to use index option in configuration

### LINKED ISSUE #49780: DOCS missing for replacing index.html with custom one in Angular configuration
### Describe the problem that you experienced

There is an option in the Angular project configuration to set a different index.html file instead of the default one. It is very handy when you want to have different scripts in HTML per environment, like GTM, for example.
Unfortunately, it is not mentioned anywhere in Angular docs.
So the proposal is to add it in Angular DOCS.

### Enter the URL of the topic with the problem

_No response_

### Describe what you were looking for in the documentation

I was looking for an option that would replace the default index.html file with a different one during build/serve time

### Describe the actions that led you to experience the problem

_No response_

### Describe what you want to experience that would fix the problem

_No response_

### Add a screenshot if that helps illustrate the problem

_No response_

### If this problem caused an exception or error, please paste it here

_No response_

### If the problem is browser-specific, please specify the device, OS, browser, and version

_No response_

### Provide any additional information here in as much as detail as you can

```true
PR for this option:
https://github.com/angular/angular-cli/pull/15010
More Info:
https://stackoverflow.com/questions/59324204/how-to-replace-index-html-in-angular-8-3-depending-on-environment
```

### PR DESCRIPTION
**Pull request by ghost**
Content is used mostly from 
https://github.com/angular/angular-cli/blob/c6c045b280edf8a3990e90c689b2e8862d298010/packages/angular_devkit/build_angular/src/builders/browser/schema.json#L369-L395

closes #49780

### CODE REVIEW

File: aio/content/guide/workspace-config.md [hunk 1]
```diff
@@ -514,6 +514,24 @@ These are useful if you only want source maps to map error stack traces in error
 
 </div>
 
+### Index configuration option
+
+Configures the generation of the application's HTML index.
```
Comment (JeanMeche) [inline aio/content/guide/workspace-config.md:519]: You should speficy that this option is only for `@angular-devkit/build-angular:browser` and not the others.

Comment (lukaonik) [reply aio/content/guide/workspace-config.md:519]: good call! is it worth it to put it as a note or basic statement ?

Comment (alan-agius4) [reply aio/content/guide/workspace-config.md:519]: You can add something like `(Not available in test section.)` in https://github.com/angular/angular/blob/84cd7bb474ef5abbf069d9d5fa3f7f0d0193780f/aio/content/guide/workspace-config.md#extra-build-and-test-options

File: aio/content/guide/workspace-config.md [hunk 2]
```diff
@@ -514,6 +514,24 @@ These are useful if you only want source maps to map error stack traces in error
...
+
+The `index` option can have either simple path string value, or object value with specific fields.
+
+
+#### string
```
Comment (alan-agius4) [inline aio/content/guide/workspace-config.md:524]: Can you do something like https://github.com/angular/angular/blob/84cd7bb474ef5abbf069d9d5fa3f7f0d0193780f/aio/content/guide/workspace-config.md#optimization-configuration instead please?

File: aio/content/guide/workspace-config.md (hunk 1 above)
Comment (lukaonik) [reply aio/content/guide/workspace-config.md:523]: I put it below the fileReplacement row

File: aio/content/guide/workspace-config.md (hunk 2 above)
Comment (lukaonik) [reply aio/content/guide/workspace-config.md]: I used the table style, I hope it's better

File: aio/content/guide/workspace-config.md [hunk 3]
```diff
@@ -243,6 +243,8 @@ Some extra options can only be set through the configuration file, either by dir
...
 | `budgets`                  | Default size-budget type and thresholds for all or parts of your application. You can configure the builder to report a warning or an error when the output reaches or exceeds a threshold size. See [Configure size budgets](guide/build#configure-size-budgets). \(Not available in `test` section.\) |
 | `fileReplacements`         | An object containing files and their compile-time replacements. See more in [Configure target-specific file replacements](guide/build#configure-target-specific-file-replacements).                                                                                                                     |
+ `index`                    | Configures the generation of the application's HTML index. See more in [Index configuration option](#index-config-option). (Only available in `browser` section )                                                                                                         |
```
Comment (alan-agius4) [inline aio/content/guide/workspace-config.md:246]:
```suggestion
 `index`                    | Configures the generation of the application's HTML index. See more in [Index configuration](#index-config). \(Only available in `browser` section.\)                                                                                                         |
```

File: aio/content/guide/workspace-config.md [hunk 4]
```diff
@@ -514,6 +516,31 @@ These are useful if you only want source maps to map error stack traces in error
 
 </div>
 
+<a id="index-config-option"></a>
+
+### Index configuration option
```
Comment (alan-agius4) [inline aio/content/guide/workspace-config.md:521]: For consistency I suggest

```suggestion
<a id="index-config"></a>

### Index configuration
```

File: aio/content/guide/workspace-config.md [hunk 5]
```diff
@@ -514,6 +516,31 @@ These are useful if you only want source maps to map error stack traces in error
...
+#### Object option
+
+The object value contains two fields `input` and `output`.
+
+| Fields   | Details                                                                                                                                                                          |
+| :------- | :------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| `input`  | The path of a file to use for the application's generated HTML index.                                                                                                            |
+| `output` | The output path of the application's generated HTML index file. The full provided path will be used and will be considered relative to the application's configured output path. |
+
+
```
Comment (alan-agius4) [inline aio/content/guide/workspace-config.md:543]:
```suggestion
```suggestion
The `index` option can be either a String or an Object for more fine-tune configuration.

When supplying the value as a String the filename of the specified path will be used for the generated file and will be created in the root of the application's configured output path.

#### Index options
| Options  | Details                                                                                                                                                                          | Value type | Default value |
|:---      |:---                                                                                                                                                                              |:---        |:---           |
| `input`  | The path of a file to use for the application's generated HTML index.                                                                                                            | `string`   |               |
| `output` | The output path of the application's generated HTML index file. The full provided path will be used and will be considered relative to the application's configured output path. | `string`   | `index.html`  |
```

```

File: aio/content/guide/workspace-config.md [hunk 6]
```diff
@@ -243,6 +243,8 @@ Some extra options can only be set through the configuration file, either by dir
 | `scripts`                  | An object containing JavaScript script files to add to the global context of the project. The scripts are loaded exactly as if you had added them in a `<script>` tag inside `index.html`. See more in the [Styles and scripts configuration](#style-script-config) section.                            |
```
Comment (alan-agius4) [inline aio/content/guide/workspace-config.md:243]: Can you also please update the following
```diff

## Complex configuration values
-  The `assets`, `styles`, and `scripts` options can have either simple path string values, or object values with specific fields.
+  The `assets`,`index`, `styles`, and `scripts` options can have either simple path string values, or object values with specific fields.
```

**Review by alan-agius4 — CHANGES_REQUESTED** [review]
Thanks for this a couple of improvements.

From the commits, can you also please remove the `revert` Thanks

**Comment by lukaonik** [comment]
@alan-agius4 I cleaned up the commit history, and applied suggestions. I hope that's what you asked for ;)

File: aio/content/guide/workspace-config.md [hunk 7]
```diff
@@ -243,12 +243,14 @@ Some extra options can only be set through the configuration file, either by dir
...
 | `fileReplacements`         | An object containing files and their compile-time replacements. See more in [Configure target-specific file replacements](guide/build#configure-target-specific-file-replacements).                                                                                                                     |
+`index`                    | Configures the generation of the application's HTML index. See more in [Index configuration](#index-config). \(Only available in `browser` section.\)                                                                                                         |                                                                                                        |                                                                                                      |
+
 
 <a id="complex-config"></a>
 
 ## Complex configuration values
 
-The `assets`, `styles`, and `scripts` options can have either simple path string values, or object values with specific fields.
+The `assets`,`index`, `styles`, and `scripts` options can have either simple path string values, or object values with specific fields.
```
Comment (alan-agius4) [inline aio/content/guide/workspace-config.md:253]:
```suggestion
The `assets`, `index`, `styles`, and `scripts` options can have either simple path string values, or object values with specific fields.
```

File: aio/content/guide/workspace-config.md [hunk 8]
```diff
@@ -243,12 +243,14 @@ Some extra options can only be set through the configuration file, either by dir
...
 | `budgets`                  | Default size-budget type and thresholds for all or parts of your application. You can configure the builder to report a warning or an error when the output reaches or exceeds a threshold size. See [Configure size budgets](guide/build#configure-size-budgets). \(Not available in `test` section.\) |
 | `fileReplacements`         | An object containing files and their compile-time replacements. See more in [Configure target-specific file replacements](guide/build#configure-target-specific-file-replacements).                                                                                                                     |
+`index`                    | Configures the generation of the application's HTML index. See more in [Index configuration](#index-config). \(Only available in `browser` section.\)                                                                                                         |                                                                                                        |                                                                                                      |
```
Comment (alan-agius4) [inline aio/content/guide/workspace-config.md:246]:
```suggestion
`index`                    | Configures the generation of the application's HTML index. See more in [Index configuration](#index-config). \(Only available in `browser` section.\)                                                                                                         |                                                                                                        |                                                                                                      
```

File: aio/content/guide/workspace-config.md [hunk 9]
```diff
@@ -514,6 +516,22 @@ These are useful if you only want source maps to map error stack traces in error
 
 </div>
 
+<a id="index-config"></a>
+
+### Index configuration
+
+Configures the generation of the application's HTML index.
+
+The `index` option can be either a String or an Object for more fine-tune configuration.
```
Comment (alan-agius4) [inline aio/content/guide/workspace-config.md:525]:
```suggestion
The `index` option can be either a String or an Object for more fine-tune configuration.

```

File: aio/content/guide/workspace-config.md [hunk 10]
```diff
@@ -514,6 +516,22 @@ These are useful if you only want source maps to map error stack traces in error
...
+When supplying the value as a String the filename of the specified path will be used for the generated file and will be created in the root of the application's configured output path.
+#### Index options
```
Comment (alan-agius4) [inline aio/content/guide/workspace-config.md:527]:
```suggestion

#### Index options
```

File: aio/content/guide/workspace-config.md [hunk 11]
```diff
@@ -514,6 +516,22 @@ These are useful if you only want source maps to map error stack traces in error
...
+
```
Comment (alan-agius4) [inline aio/content/guide/workspace-config.md:528]:
```suggestion
```

**Review by alan-agius4 — CHANGES_REQUESTED** [review]
Thanks for this, just a couple of styling NITs.

**Comment by lukaonik** [comment]
I hope that's the last one :D 🙏

**Comment by alan-agius4** [comment]
@lukaonik, it looks like the last commit is not valid due to the multiple `fixup!`.

`fixup! fixup! docs: …` -> `fixup! docs: …`

**Comment by lukaonik** [comment]
ah right, sec

**Comment by lukaonik** [comment]
I think it should be fine now @alan-agius4

**Review by alan-agius4 — APPROVED** [review]
LGTM, thanks for your contribution.

**Comment by lukaonik** [comment]
@alan-agius4 thank you a lot for the support! This is my first PR to Angular 👏

**Comment by alan-agius4** [comment]
@lukaonik, you are very welcome. 😀

**Comment by jessicajaniuk** [comment]
This PR was merged into the repository by commit a29c51b.

**Pull request closed without merging**